From: Hannes Reinecke <hare@suse.de>
Date: Thu, 20 Mar 2008 23:53:48 +0100
Subject: [PATCH] [libmultipath] Update discovery to work with new sysfs layout

With the new sysfs layout the parent device of a block device
it 'block', and only the parent of this is the 'real' parent.
---
 libmultipath/discovery.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/libmultipath/discovery.c b/libmultipath/discovery.c
index c842eb0..2600c1e 100644
--- a/libmultipath/discovery.c
+++ b/libmultipath/discovery.c
@@ -546,6 +546,9 @@ sysfs_pathinfo(struct path * pp)
 	if (!parent)
 		parent = pp->sysdev;
 
+	if (!strncmp(parent->kernel, "block",5))
+		parent = sysfs_device_get_parent(parent);
+
 	condlog(3, "%s: subsystem = %s", pp->dev, parent->subsystem);
 
 	if (!strncmp(parent->subsystem, "scsi",4))
-- 
